zhaquirks/eurotronic/__init__.py

```diff
@@ -0,0 +1,160 @@
+"""Eurotronic devices."""
+
+import logging
+
+import zigpy.types as types
+from zigpy.quirks import CustomCluster
+from zigpy.zcl import foundation
+from zigpy.zcl.clusters.hvac import Thermostat
+
+
+EUROTRONIC = "Eurotronic"
+
+THERMOSTAT_CHANNEL = "thermostat"
+
+MANUFACTURER = 0x1037  # 4151
+
+OCCUPIED_HEATING_SETPOINT_ATTR = 0x0012
+CTRL_SEQ_OF_OPER_ATTR = 0x001B
+SYSTEM_MODE_ATTR = 0x001C
+
+TRV_MODE_ATTR = 0x4000
+SET_VALVE_POS_ATTR = 0x4001
+ERRORS_ATTR = 0x4002
+CURRENT_TEMP_SETPOINT_ATTR = 0x4003
+HOST_FLAGS_ATTR = 0x4008
+
+
+# Host Flags
+# unknown (defaults to 1)       = 0b00000001 # 1
+MIRROR_SCREEN_FLAG = 0b00000010  # 2
+BOOST_FLAG = 0b00000100  # 4
+# unknown                       = 0b00001000 # 8
+CLR_OFF_MODE_FLAG = 0b00010000  # 16
+SET_OFF_MODE_FLAG = 0b00100000  # 32, reported back as 16
+# unknown                       = 0b01000000 # 64
+CHILD_LOCK_FLAG = 0b10000000  # 128
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ThermostatCluster(CustomCluster, Thermostat):
+    """Thermostat cluster."""
+
+    cluster_id = Thermostat.cluster_id
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+
+        # manufacturer specific attributes
+        self.attributes.update(
+            {
+                TRV_MODE_ATTR: ("trv_mode", types.enum8),
+                SET_VALVE_POS_ATTR: ("set_valve_position", types.uint8_t),
+                ERRORS_ATTR: ("errors", types.uint8_t),
+                CURRENT_TEMP_SETPOINT_ATTR: (
+                    "current_temperature_setpoint",
+                    types.int16s,
+                ),
+                HOST_FLAGS_ATTR: ("host_flags", types.uint24_t),
+            }
+        )
+        self.attridx = {
+            attrname: attrid for attrid, (attrname, datatype) in self.attributes.items()
+        }
```

Comment:
```suggestion
    attributes = {
        TRV_MODE_ATTR: ("trv_mode", types.enum8),
        SET_VALVE_POS_ATTR: ("set_valve_position", types.uint8_t),
        ERRORS_ATTR: ("errors", types.uint8_t),
        CURRENT_TEMP_SETPOINT_ATTR: ("current_temperature_setpoint", types.int16s),
        HOST_FLAGS_ATTR: ("host_flags", types.uint24_t),
    }
    attributes.update(Thermostat.attributes)
```

You could do this as well, then there's no need for custom init or updating attridx

Reply:
Thanks, I really struggled with that one, this looks so much cleaner!